mne/preprocessing/tests/test_ica.py

```diff
@@ -735,21 +735,6 @@ def f(x, y):
         pytest.raises(RuntimeError, ica.find_bads_ecg, raw)
 
 
-@requires_sklearn
-@pytest.mark.parametrize("method", ["fastica", "picard"])
-def test_run_ica(method):
-    """Test run_ica function."""
-    _skip_check_picard(method)
-    raw = read_raw_fif(raw_fname).crop(1.5, stop).load_data()
-    params = []
-    params += [(None, -1, slice(2), [0, 1])]  # varicance, kurtosis idx
-    params += [(None, 'MEG 1531')]  # ECG / EOG channel params
-    for idx, ch_name in product(*params):
-        run_ica(raw, n_components=2, start=0, stop=0.5, start_find=0,
-                stop_find=5, ecg_ch=ch_name, eog_ch=ch_name, method=method,
-                skew_criterion=idx, var_criterion=idx, kurt_criterion=idx)
```

Comment:
Then it would be nice to refactor this test to use the new pattern, unless you're confident that the existing `detect_artifacts` tests adequately cover the use cases that are iterated over here.

If you do keep this test, it might be nice to `parametrize` (new way) instead of using the `product` (old way)

Reply:
> would be nice to refactor this test to use the new pattern

Not sure what you mean by that..?

Reply:
The new pattern suggested by the deprecation, i.e. replace the old way with the new way:
```
ica = ICA(...)
ica.fit(...)
ica.detect_artifacts(...)
```

Reply:
This test is parametrized over 2 values of `ch_name`, 4 of `idx`, and 3 `method`s.  Do we really need to test that the warning is raised 24 times?  I'd make a separate test function `test_run_ica_deprecation` or similar that just tests the deprecation once, and have a separate function that uses the new way (as @larsoner suggests) and is parametrized.